File: update_imports.py
import os

root = '/home/lshdainty/study/porest-hr-front/src'
count = 0
for dirpath, dirnames, filenames in os.walk(root):
    if '/components/shadcn' in dirpath or '/shared/ui/shadcn' in dirpath:
        continue
    for fn in filenames:
        if fn.endswith(('.ts', '.tsx')):
            fp = os.path.join(dirpath, fn)
            with open(fp, 'r') as f:
                content = f.read()
            if '@/components/shadcn/' in content:
                new_content = content.replace('@/components/shadcn/', '@/shared/ui/shadcn/')
                with open(fp, 'w') as f:
                    f.write(new_content)
                count += 1
print(f'Updated {count} files')

File: update_imports.js
const fs = require('fs')
const path = require('path')

function walk(dir) {
  let results = []
  const list = fs.readdirSync(dir)
  list.forEach(file => {
    file = path.join(dir, file)
    const stat = fs.statSync(file)
    if (stat && stat.isDirectory()) {
      if (!file.includes('/components/shadcn') && !file.includes('/shared/ui/shadcn')) {
        results = results.concat(walk(file))
      }
    } else {
      if (file.endsWith('.ts') || file.endsWith('.tsx')) results.push(file)
    }
  })
  return results
}

const files = walk('src')
let count = 0
files.forEach(fp => {
  const content = fs.readFileSync(fp, 'utf8')
  if (content.includes('@/components/shadcn/')) {
    fs.writeFileSync(fp, content.replace(/@\/components\/shadcn\//g, '@/shared/ui/shadcn/'))
    count++
  }
})
console.log(count + ' files updated')
